test/lib/ansible_test/_internal/docker_util.py

```diff
@@ -507,7 +507,7 @@ def get_docker_hostname() -> str:
 
     if docker_host and docker_host.startswith(('tcp://', 'ssh://')):
         try:
-            hostname = urllib.parse.urlparse(docker_host)[1].split(':')[0]
+            hostname = urllib.parse.urlparse(docker_host).hostname
```

Comment:
What value for `DOCKER_HOST` are you using that this failed?

Reply:
The current parsing doesn't strip the username, so I was getting `user@192.168.0.100` from `[1]`